presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlock.java

```diff
@@ -142,9 +143,11 @@ public Block copyRegion(int positionOffset, int length)
             throw new IndexOutOfBoundsException("Invalid position " + positionOffset + " in block with " + positionCount + " positions");
         }
 
-        Slice newOffsets = Slices.copyOf(offsets, positionOffset * SIZE_OF_INT, (length + 1) * SIZE_OF_INT);
-        Slice newValueIsNull = Slices.copyOf(valueIsNull, positionOffset, length);
-        return new VariableWidthBlock(length, slice, newOffsets, newValueIsNull);
+        List<Integer> positions = new ArrayList<>(length);
```

Comment:
It looks like this is going to allocate a lot of objects, and `copyRegion` may be called in hot locations like `AbstractArrayBlock.getSingleValueBlock`, so I'd optimize it to use `Slices.copyOf` for the `slice`